src/SharpLearning.Backend.TensorFlow.Test/TensorFlowRawMnistDeepTest.cs

```diff
@@ -119,32 +148,87 @@ public void MnistDeep()
 
                             Log($"Step {i} Accuracy {train_acc}");
                         }
-
+#if USE_OLD_READER
                         TFTensor[] inputValues = new[] { new TFTensor(inputBatch), new TFTensor(labelBatch), train_dropout_keep_prob };
+#else
+                        TFTensor[] inputValues = new[] {
+                            TFTensor.FromBuffer(new TFShape(batchSize, FeatureCount), inputBatch, 0, inputBatch.Length),
+                            TFTensor.FromBuffer(new TFShape(batchSize, ClassCount), labelBatch, 0, labelBatch.Length),
+                            train_dropout_keep_prob};
+#endif
                         TFTensor[] outputValues = session.Run(inputs, inputValues, outputs,
                             targets, runMetaData, runOptions, trainStatus);
 
                         trainStatus.Raise();
                     }
+                    var trainTime_ms = s.Elapsed.TotalMilliseconds;
+                    s.Restart();
 
                     // Test trained model
+#if USE_OLD_READER
                     var testReader = mnist.GetTestReader();
                     var (testImages, testLabels) = testReader.All();
-
                     TFTensor evaluatedAccuracy = session.GetRunner()
                         .AddInput(x, testImages)
                         .AddInput(y_, testLabels)
-                        .AddInput(keep_prob, test_dropout_keep_prob) // DOESN*T WORK!??!
+                        .AddInput(keep_prob, test_dropout_keep_prob)
                         .Run(accuracy);
-
                     float acc = (float)evaluatedAccuracy.GetValue();
+#else
+                    // Batch evaluation is faster (2x) and uses A LOT less memory, not batched == 6GB, batched == 240MB!
```

Comment:
@mdabros huge improvement when doing evaluation batched. Code is pretty messed up  right now, and deep still does not give the same as python. I have now removed the validation stuff in both python and C#.